### PR TITLE
Absolute tolerances

### PR DESCRIPTION
**Pull request by andrewbriand**

### CODE REVIEW

File: .vs/ProjectSettings.json [hunk 1]
```diff
@@ -0,0 +1,3 @@
+{
```
Comment (andrewbriand) [inline .vs/ProjectSettings.json]: We should probably gitignore the .vs folder

File: source/CVODEIntegrator.cpp [hunk 2]
```diff
@@ -281,10 +324,188 @@ namespace rr
...
+		// update the tolerance
+		CVODEIntegrator::setValue("absolute_tolerance", v);
+	}
+
+
+	void CVODEIntegrator::setConcentrationTolerance(const Variant& value) {
+
+		uint ncomp = mModel->getNumCompartments();
+
+		double* volumes = (double*)calloc(ncomp, sizeof(double));
```
Comment (andrewbriand) [inline source/CVODEIntegrator.cpp:409]: Does this memory ever get freed?

File: source/CVODEIntegrator.cpp [hunk 3]
```diff
@@ -281,10 +324,188 @@ namespace rr
...
+			{
+				// scalar concentration tolerance
+				// need to be converted to vector tolerance since speices might have various compartment sizes
+				double abstol = value.convert<double>();
+
+				for (int i = 0; i < mModel->getNumFloatingSpecies(); i++) {
+					// get the compartment volume of each species
+					int index = mModel->getCompartmentIndexForFloatingSpecies(i);
+					if (volumes[index] == 0) {
+						// when the compartment volume is 0, simply set the amount tolerance as the volume tolerance
```
Comment (andrewbriand) [inline source/CVODEIntegrator.cpp:433]: Should we throw an exception if a compartment's volume is 0?

File: source/CVODEIntegrator.cpp [hunk 4]
```diff
@@ -281,10 +324,188 @@ namespace rr
...
+		}
+
+		// update the tolerance
+		CVODEIntegrator::setValue("absolute_tolerance", v);
+	}
+
+	vector<double> CVODEIntegrator::getConcentrationTolerance() {
+		uint ncomp = mModel->getNumCompartments();
+
+		double* volumes = (double*)calloc(ncomp, sizeof(double));
```
Comment (andrewbriand) [inline source/CVODEIntegrator.cpp:529]: Does this memory ever get freed?

Comment (faye-u) [reply source/CVODEIntegrator.cpp:529]: Fixed it! Thank you!

File: source/CVODEIntegrator.cpp (hunk 2 above)
Comment (faye-u) [reply source/CVODEIntegrator.cpp:409]: Fixed it! Thank you!

File: source/CVODEIntegrator.cpp (hunk 3 above)
Comment (faye-u) [reply source/CVODEIntegrator.cpp:433]: No need to throw an exception for now.

**Pull request merged**